syntax = "proto3";

package bloock;

option go_package = "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto";
option java_package = "com.bloock.sdk.bridge.proto";

import "keys_entities.proto";

enum HashAlg {
    SHA_256 = 0;
    KECCAK_256 = 1;
    POSEIDON = 2;
}

message Signer {
    optional LocalKey local_key = 1;
    optional ManagedKey managed_key = 2;
    optional LocalCertificate local_certificate = 3;
    optional ManagedCertificate managed_certificate = 4;
    optional HashAlg hash_alg = 5;
}

message Signature {
    string signature = 1;
    string alg = 2;
    string kid = 3;
    string message_hash = 4;
    optional string subject = 5;
    optional string hash_alg = 6;
}